pkg/noderesourcetopology/cache/discardreserved_test.go

```diff
@@ -0,0 +1,172 @@
+
+import (
+	"reflect"
+	"testing"
+
+	topologyv1alpha2 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+	faketopologyv1alpha2 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned/fake"
+	topologyinformers "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestDiscardReservedNodesGetNRTCopy(t *testing.T) {
+	fakeClient := faketopologyv1alpha2.NewSimpleClientset()
+	fakeInformer := topologyinformers.NewSharedInformerFactory(fakeClient, 0).Topology().V1alpha2().NodeResourceTopologies()
+
+	nrtCache := NewDiscardReserved(fakeInformer.Lister())
+	var nrtObj *topologyv1alpha2.NodeResourceTopology
+	nrtObj, _ = nrtCache.GetCachedNRTCopy("node1", &corev1.Pod{})
+	if nrtObj != nil {
+		t.Fatalf("non-empty object from empty cache")
+	}
+
+	nodeTopologies := []*topologyv1alpha2.NodeResourceTopology{
+		{
+			ObjectMeta:       metav1.ObjectMeta{Name: "node1"},
+			TopologyPolicies: []string{string(topologyv1alpha2.SingleNUMANodeContainerLevel)},
+			Zones: topologyv1alpha2.ZoneList{
+				{
+					Name: "node-0",
+					Type: "Node",
+					Resources: topologyv1alpha2.ResourceInfoList{
+						MakeTopologyResInfo(cpu, "20", "4"),
+						MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						MakeTopologyResInfo(nicResourceName, "30", "10"),
+					},
+				},
+				{
+					Name: "node-1",
+					Type: "Node",
+					Resources: topologyv1alpha2.ResourceInfoList{
+						MakeTopologyResInfo(cpu, "30", "8"),
+						MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						MakeTopologyResInfo(nicResourceName, "30", "10"),
+					},
+				},
+			},
+		},
+	}
+	for _, obj := range nodeTopologies {
+		fakeInformer.Informer().GetStore().Update(obj)
+	}
+
+	nrtObj, ok := nrtCache.GetCachedNRTCopy("node1", &corev1.Pod{})
+	if !reflect.DeepEqual(nrtObj, nodeTopologies[0]) {
+		t.Fatalf("unexpected object from cache\ngot: %s\nexpected: %s\n", dumpNRT(nrtObj), dumpNRT(nodeTopologies[0]))
+	}
+
+	if !ok {
+		t.Fatalf("expecting GetCachedNRTCopy to return true not false")
+	}
+}
+
+func TestDiscardReservedNodesGetNRTCopyFails(t *testing.T) {
+	nrtCache := DiscardReserved{
+		reservationMap: map[string]map[types.UID]bool{
+			"node1": {
+				types.UID("some-uid"): true,
+			},
+		},
+	}
+
+	nrtObj, ok := nrtCache.GetCachedNRTCopy("node1", &corev1.Pod{})
+	if ok {
+		t.Fatal("expected false\ngot true\n")
+	}
+	if nrtObj != nil {
+		t.Fatalf("non-empty object")
+	}
+}
+
+func TestDiscardReservedNodesReserveNodeResources(t *testing.T) {
+	nrtCache := DiscardReserved{
+		reservationMap: make(map[string]map[types.UID]bool),
+	}
+
+	nrtCache.ReserveNodeResources("node1", &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pod",
+			Namespace: "test",
+			UID:       "some-uid",
+		},
+	})
+	nodePods, ok := nrtCache.reservationMap["node1"]
```

Comment:
let's improve the tests later on
1. adding (more) integration tests
2. avoiding to access private fields/methods